src/Listener/RelatedModelsListener.php

```diff
@@ -51,6 +52,27 @@ public function beforePaginate(Event $event)
     }
 
     /**
+     * Automatically parse and contain related table classes
+     *
+     * @param \Cake\Event\Event $event Before find event
+     * @return void
+     */
+    public function beforeFind(Event $event)
+    {
+        $contained = $event->subject()->query->contain();
```

Comment:
Fantastic idea! Should make the code simpler and clearer, I have implemented it and am pushing it in a bit.